### PR TITLE
Closes #544: Transition map view controller in from top

### LINKED ISSUE #544: Transition map view controller in from top

### PR DESCRIPTION
**Pull request by thebnich**
Copied from `FilterViewController`, using different views the for animations.

### CODE REVIEW

File: Prox/Prox/Map View/MapViewController.swift [hunk 1]
```diff
@@ -77,31 +81,50 @@ class MapViewController: UIViewController {
...
+        for subview in [container, mapView, searchButton, placeFooter, closeButton] as [UIView] {
             view.addSubview(subview)
         }
 
+        container.snp.makeConstraints { make in
+            make.leading.trailing.equalToSuperview()
+
+            showMapConstraints = [
+                make.top.equalToSuperview().constraint,
+                make.bottom.equalTo(bottomLayoutGuide.snp.top).constraint,
```
Comment (mcomella) [inline Prox/Prox/Map View/MapViewController.swift]: Why bottom layout guide?

File: Prox/Prox/Map View/MapViewController.swift [hunk 2]
```diff
@@ -131,22 +154,13 @@ class MapViewController: UIViewController {
     }
 
     @objc private func closeWithButton() {
-        resetMapToUserLocation(shouldAnimate: true)
-
-        // HACK: (no UX input) Delay the dismiss so the user can watch the animation and see what happens.
-        // TODO: we should change the delay depending on how far we animate.
-        // Note: I tried to add a listener for the end of the animation to dismiss but it complicated the code too much
-        // so here's this hack.
-        DispatchQueue.main.asyncAfter(deadline: DispatchTime.now() + DispatchTimeInterval.milliseconds(500)) {
```
Comment (mcomella) [inline Prox/Prox/Map View/MapViewController.swift]: As per irl, I don't think we should rm this.

Comment (thebnich) [inline Prox/Prox/Map View/MapViewController.swift]: Talking to @antlam, he agrees that the return-to-position effect isn't worth the lag the user feels after pressing the close button, so keeping rm'ed.

File: Prox/Prox/Map View/MapViewController.swift (hunk 1 above)
Comment (thebnich) [inline Prox/Prox/Map View/MapViewController.swift]: Just copied from [what we were using before](https://github.com/mozilla-mobile/prox/pull/630/files#diff-8364cdee0b6d84d4da354f4b18cc16b5L116). I'll just change this to bottom.

**Pull request merged**